src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -1040,51 +1023,55 @@ private async Task<string> UploadPackageToStorage(Stream package, string blobNam
+                    }
+
+                    await blobContainerClient.CreateIfNotExistsAsync();
                 }
                 catch (Exception ex)
                 {
                     if (StaticSettings.IsDebug)
                     {
                         ColoredConsole.Error.WriteLine(ErrorColor(ex.ToString()));
                     }
-                    throw new CliException($"Error creating a Blob container reference. Please make sure your connection string in \"AzureWebJobsStorage\" is valid");
-                }
 
-                var blob = blobContainer.GetBlockBlobReference(blobName);
-                using (var progress = new StorageProgressBar($"Uploading {Utilities.BytesToHumanReadable(package.Length)}", package.Length))
-                {
-                    await blob.UploadFromStreamAsync(package,
-                        AccessCondition.GenerateEmptyCondition(),
-                        new BlobRequestOptions(),
-                        new OperationContext(),
-                        progress,
-                        new CancellationToken());
+                    throw new CliException($"Error creating a blob container client. Please make sure the account name in \"{Constants.AzureWebJobsStorageAccountName}\" or the connection string in \"{Constants.AzureWebJobsStorage}\" is valid.");
                 }
 
-                var cloudMd5 = blob.Properties.ContentMD5;
+                var blob = blobContainerClient.GetBlobClient(blobName);
+                using var progress = new StorageProgressBar($"Uploading {Utilities.BytesToHumanReadable(package.Length)}", package.Length);
+                var blobInfo = await blob.UploadAsync(package, new BlobUploadOptions { ProgressHandler = progress });
+
+                var cloudMd5 = Convert.ToBase64String(blobInfo.Value.ContentHash);
 
                 if (!cloudMd5.Equals(packageMD5))
                 {
                     throw new CliException("Upload failed: Integrity error: MD5 hash mismatch between the local copy and the uploaded copy.");
                 }
 
-                var sasConstraints = new SharedAccessBlobPolicy();
-                sasConstraints.SharedAccessStartTime = DateTimeOffset.UtcNow.AddMinutes(-5);
-                sasConstraints.SharedAccessExpiryTime = DateTimeOffset.UtcNow.AddYears(10);
-                sasConstraints.Permissions = SharedAccessBlobPermissions.Read;
-
-                var blobToken = blob.GetSharedAccessSignature(sasConstraints);
+                var sasConstraints = new BlobSasBuilder
+                {
+                    StartsOn = DateTimeOffset.UtcNow.AddMinutes(-5),
+                    ExpiresOn = DateTimeOffset.UtcNow.AddYears(10),
+                };
+                sasConstraints.SetPermissions(BlobAccountSasPermissions.Read);
 
-                return blob.Uri + blobToken;
+                return blob.GenerateSasUri(sasConstraints).ToString();
```

Comment:
Again, to allow disabling Shared Keys on the storage account, return blob.Uri in case managed identity is used.